python/ml4ir/base/features/feature_config.py

```diff
@@ -465,13 +473,11 @@ def parse_config(
     tfrecord_type: str, feature_config, logger: Optional[Logger] = None
 ) -> FeatureConfig:
     if feature_config.endswith(".yaml"):
-        feature_config = file_io.read_yaml(feature_config)
-        if logger:
-            logger.info("Reading feature config from YAML file : {}".format(feature_config))
```

Comment:
I am just wondering here. Not for this PR. 
But `if logger` repeats a lot in ml4ir. Why not agree we always need a logger, and instantiate one form the beginning? Or decorate key functions/classes to avoid such checks.

Reply:
Yeah. Let me file an issue for this. It looks to me like tech debt building.

Reply:
I agree - we should just always have a Logger - if none is configured, use a dummy / noop logger or something.